File: DailyChallenges/CPP/LeetCode_1833.cpp
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        sort(costs.begin(), costs.end());
        int ans = 0;
        for (int c: costs){
            if (c > coins) return ans;
            ++ans;
            coins -= c;
        }
        return ans;
    }
};

File: DailyChallenges/Scala/LeetCode_1833.scala
object Solution {
    def maxIceCream(costs: Array[Int], coins: Int): Int = {
        var count = 0
        costs.sorted.foldLeft(coins)( (left, i) => if (left >= i) { count += 1; (left - i)} else return count)
        count // case when you got a large amount of money can buy all
    }
}
